File: 03_A/VB/CSV_Reader/OnlineMean.vb
Public Class OnlineMean
    Private currentAvg As Double = 0
    Private index As Integer = 0

    Public Function GetAvg() As Double
        Return currentAvg
    End Function

    Public Sub NewValue(ByVal val As Double)
        index += 1
        currentAvg += (val - currentAvg) / index
    End Sub
End Class

File: 03_A/C#/CSV_Reader/OnlineMean.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace CSV_Reader {
    class OnlineMean {
        private double currentAvg = 0;
        private int index = 0;

        public double GetAvg() { return currentAvg; }

        public void NewValue (double val) {
            index++;
            currentAvg += (val - currentAvg) / index;
        }
    }
}
